airflow/contrib/operators/databricks_operator.py

```diff
@@ -0,0 +1,196 @@
+        self.json = json or {}
+        self.spark_jar_task = spark_jar_task
+        self.notebook_task = notebook_task
+        self.new_cluster = new_cluster
+        self.existing_cluster_id = existing_cluster_id
+        self.libraries = libraries
+        self.run_name = run_name
+        self.timeout_seconds = timeout_seconds
+        self.databricks_conn_id = databricks_conn_id
+        self.polling_period_seconds = polling_period_seconds
+        self.kwargs = kwargs
+
+    def _get_merged_parameters(self):
+        merged = self.json.copy()
+        if self.spark_jar_task is not None:
+            merged['spark_jar_task'] = self.spark_jar_task
+        if self.notebook_task is not None:
+            merged['notebook_task'] = self.notebook_task
+        if self.new_cluster is not None:
+            merged['new_cluster'] = self.new_cluster
+        if self.existing_cluster_id is not None:
+            merged['existing_cluster_id'] = self.existing_cluster_id
+        if self.libraries is not None:
+            merged['libraries'] = self.libraries
+        if self.run_name is not None:
+            merged['run_name'] = self.run_name
+        if self.timeout_seconds is not None:
+            merged['timeout_seconds'] = self.timeout_seconds
+        if 'run_name' not in merged:
+            merged['run_name'] = self.run_name or self.kwargs['task_id']
+        return merged
+
+    def _log_run_page_url(self, url):
+        logging.info('View run status, Spark UI, and logs at {}'.format(url))
+
+    def get_hook(self):
+        return DatabricksHook(self.databricks_conn_id)
+
+    def execute(self, context):
+        hook = self.get_hook()
+        run_id = hook.submit_run(self._get_merged_parameters())
+        run_page_url = hook.get_run_page_url(run_id)
+        logging.info(LINE_BREAK)
+        logging.info('Run submitted with run_id: {}'.format(run_id))
+        self._log_run_page_url(run_page_url)
+        logging.info(LINE_BREAK)
+        while True:
+            run_state = hook.get_run_state(run_id)
+            if run_state.is_terminal:
+                if run_state.is_successful:
+                    logging.info('{} completed successfully.'.format(
+                        self.task_id))
+                    self._log_run_page_url(run_page_url)
+                    return
+                else:
+                    error_message = '{t} failed with terminal state: {s}'.format(
+                        t=self.task_id,
+                        s=run_state)
+                    raise AirflowException(error_message)
+            else:
+                logging.info('{t} in run state: {s}'.format(t=self.task_id,
+                                                            s=run_state))
+                self._log_run_page_url(run_page_url)
+                logging.info('Sleeping for {} seconds.'.format(
+                    self.polling_period_seconds))
+                time.sleep(self.polling_period_seconds)
```

Comment:
On last part that you might want to implement is the `on_kill`method if there is a way for user to cancel a run.